src/services/api/api.controller.ts

```diff
@@ -147,19 +152,22 @@ export class ApiController {
     },
   })
   @UseGuards(AuthGuard('jwt'))
+  @UseInterceptors(UserDetailsInterceptor)
   @Post(`/hive/linkaccount`)
-  async linkAccount(
-    @Body() data: LinkAccountPostDto,
-    @Request() req: { user: { user_id: string } },
-  ) {
-    const { user_id } = req.user; // TODO: security
+  async linkAccount(@Body() data: LinkAccountPostDto, @Request() req: unknown) {
+    const parsedRequest = parseAndValidateRequest(req, this.#logger);
     const linkedAccount = await this.linkedAccountsRepository.findOneByUserIdAndAccountName({
-      user_id: user_id,
+      user_id: parsedRequest.user.sub,
```

Comment:
Is this correct? @vaultec81

Reply:
it looks like `id` in the `user` reffers to a session id based on other parts of the code. I think this could be username in all fairness but user_id never existed on the token, will wait for vaultecs input

Reply:
It should be the user ID to my knowledge. If you send me an example value I can confirm